File: codeforces/1929D.cpp
#include "bits/stdc++.h"
#pragma GCC optimize ("O3")
#pragma GCC target ("sse4")
 
using namespace std;
 
typedef long long ll;
typedef long double ld;
typedef complex<ld> cd;
 
typedef pair<int, int> pi;
typedef pair<ll,ll> pl;
typedef pair<ld,ld> pd;
 
typedef vector<int> vi;
typedef vector<ld> vd;
typedef vector<ll> vl;
typedef vector<pi> vpi;
typedef vector<pl> vpl;
typedef vector<cd> vcd;
 
template<class T> using pq = priority_queue<T>;
template<class T> using pqg = priority_queue<T, vector<T>, greater<T>>;
 
#define FOR(i, a, b) for (int i=a; i<(b); i++)
#define F0R(i, a) for (int i=0; i<(a); i++)
#define FORd(i,a,b) for (int i = (b)-1; i >= a; i--)
#define F0Rd(i,a) for (int i = (a)-1; i >= 0; i--)
#define trav(a,x) for (auto& a : x)
#define uid(a, b) uniform_int_distribution<int>(a, b)(rng)
 
#define sz(x) (int)(x).size()
#define mp make_pair
#define pb push_back
#define f first
#define s second
#define lb lower_bound
#define ub upper_bound
#define all(x) x.begin(), x.end()
#define ins insert

const int MOD = 998244353;
vector<vi> g;
ll ans;

int dfs(int u,int p){
    if (g[u].size()==1 && g[u][0]==p)return 2;
    ll tp1=1,tp2=0;
    trav(v,g[u]){
        if(v==p)continue;
        ll tem=dfs(v,u);
        tp1*=tem;tp1%=MOD;tp2+=tem-1;
    }
    ans+=tp2;
    return tp1+1; 
}
void solve(){
    int n;cin>>n;
    g = vector<vi>(n);
    for(int i=0;i<n-1;i++){
        int u,v;cin>>u>>v;u--;v--;
        g[u].pb(v);g[v].pb(u);
    }
    ans=0;
    ll tp=dfs(0,-1);
    ans+=tp;
    cout<<ans%MOD<<"\n";
}
int main() {
    ios_base::sync_with_stdio(0);cin.tie(0);
    int t;
    cin>>t;
    while(t--)solve();
    return 0;
}

File: codeforces/1929D.py
import sys
from collections import defaultdict
sys.setrecursionlimit(400000)
MOD=998244353
t=int(input())
def solve():
    n=int(input())
    g=defaultdict(list)
    for _ in range(n-1):
        u,v=map(int,input().split())
        g[u].append(v)
        g[v].append(u)
    vis=[0]*(n+1)
    ans=0
    def rec(u):
        vis[u]=1
        if len(g[u])==1 and vis[g[u][0]]:return 2
        tp1,tp2=1,0
        for v in g[u]:
            if not vis[v]:
                tem=rec(v)
                tp1*=tem
                tp1%=MOD
                tp2+=tem-1
        nonlocal ans
        ans+=tp2
        return tp1+1
    tp=rec(1)
    ans+=tp
    print(ans%MOD)
for _ in range(t):solve()